features/workflow.feature

```diff
@@ -46,15 +50,91 @@ Feature: Workflow
 
     Scenario: Join
         Given an existing pipeline with the workflow:
-            | job     | triggers |
-            | FOO     | BAR+BAZ  |
-            | BAR+BAZ | XYZZY    |
+            | job   | requires |
+            | FOO   | ~commit  |
+            | BAR   | FOO      |
+            | BAZ   | FOO      |
+            | QUX   | BAR, BAZ |
         When the "FOO" job is started
         And the "FOO" build succeeded
         And the "BAR" job is started
         And the "BAZ" job is started
-        Then the "XYZZY" job is not started
+        Then the "QUX" job is not started
         And the "BAR" build succeeded
         And the "BAZ" build succeeded
-        Then the "XYZZY" job is started
-        And that "XYZZY" build uses the same SHA as the "FOO" build
+        Then the "QUX" job is started
+        And that "QUX" build uses the same SHA as the "FOO" build
+
+    Scenario: Branch filtering (the pipeline's branch is committed)
+        Given an existing pipeline with the workflow:
+            | job   | requires        |
+            | FOO   | ~commit         |
+            | BAR   | ~commit:staging |
+            | BAZ   | ~commit:/^.*$/  |
+        When a new commit is pushed
+        And it is against the pipeline's branch
```

Comment:
I think `pipeline's branch` wording is confusing for me
Can we call out the branch specifically by saying something like `And it is on the "master" branch`

And is there a difference between `it is against` vs `it is targeting`?

Reply:
`pipeline's branch` is reused from git-flow feature:
https://github.com/screwdriver-cd/screwdriver/blob/a37c2485340e84240bef8147d2f9bdd337c6f082/features/git-flow.feature#L37
But I agree with you that it's confusing so I fixed.

> And is there a difference between it is against vs it is targeting?

unified with `it is targeting`

Reply:
@tkyi fixed!